### PR TITLE
esn-frontend-calendar#24: Registered i18n translation tables

### LINKED ISSUE #24: [i18n] inject translations
Details [here](https://ci.linagora.com/linagora/lgs/openpaas/esn/-/wikis/Frontend-Split/Polish/i18n#code-changes-howto-solution-1-kiss)

### PR DESCRIPTION
**Pull request by huy-ta**
Resolves #24.

The corrected translation files are taken from this PR: https://ci.linagora.com/linagora/lgs/openpaas/linagora.esn.calendar/merge_requests/664

- A quick demo to show that it works:

[Demo-Calendar-i18n](https://streamable.com/nf1e09)

### CODE REVIEW

File: src/linagora.esn.calendar/app/app.config.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require('./core/application-menu-calendar.directive.js');
```
Comment (tuanlc) [inline src/linagora.esn.calendar/app/app.config.js:1]: missing `use strict`

File: src/linagora.esn.calendar/app/app.config.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  .config(injectApplicationMenuDirective)
+  .config(registerI18N);
+
+function injectApplicationMenuDirective(dynamicDirectiveServiceProvider) {
+  var dd = new dynamicDirectiveServiceProvider.DynamicDirective(true, 'application-menu-calendar', {priority: 40});
+
+  dynamicDirectiveServiceProvider.addInjection('esn-application-menu', dd);
+}
+
+// TODO: Write tests for this (https://github.com/OpenPaaS-Suite/esn-frontend-calendar/issues/24)
```
Comment (tuanlc) [inline src/linagora.esn.calendar/app/app.config.js:13]: We never write test for config blocks before ;) AFAIK

Comment (huy-ta) [reply src/linagora.esn.calendar/app/app.config.js:13]: Yes, you're right.

File: src/linagora.esn.calendar/app/app.config.js (hunk 1 above)
Comment (huy-ta) [reply src/linagora.esn.calendar/app/app.config.js:1]: Why do we need it?

**Comment by huy-ta** [comment]
> Calendar SPA contains `linagora.esn.resource` also. Did you check it?

I've just added a commit for `linagora.esn.resource`. The Chinese and Russian translations were missing though, so I added them using Google Translate.

**Comment by tuanlc** [comment]
> The Chinese and Russian translations were missing though, so I added them using Google Translate.

You can ask for Russian translation on the translation channel. But yes, maybe no response this week. ;(

**Comment by huy-ta** [comment]
> Why there is no content for `src/linagora.esn.resource/i18n/en.json` file?

Oops, forgot to press Save.

**Comment by huy-ta** [comment]
> > The Chinese and Russian translations were missing though, so I added them using Google Translate.
> 
> You can ask for Russian translation on the translation channel. But yes, maybe no response this week. ;(

These are very simple words and sentences, so I think Google Translate can do the job pretty well.

File: src/linagora.esn.calendar/app/app.config.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+require('./core/application-menu-calendar.directive.js');
+
+angular.module('esn.calendar')
+  .config(injectApplicationMenuDirective)
+  .config(registerI18N);
+
+function injectApplicationMenuDirective(dynamicDirectiveServiceProvider) {
+  var dd = new dynamicDirectiveServiceProvider.DynamicDirective(true, 'application-menu-calendar', {priority: 40});
```
Comment (piavgh) [inline src/linagora.esn.calendar/app/app.config.js:10]: This is new code so I guess we can use `const` here

Comment (piavgh) [reply src/linagora.esn.calendar/app/app.config.js:10]: And I'm OK with other files

Comment (huy-ta) [reply src/linagora.esn.calendar/app/app.config.js:10]: Well this part of the code will be removed due to https://github.com/OpenPaaS-Suite/esn-frontend-calendar/pull/43.

Comment (piavgh) [reply src/linagora.esn.calendar/app/app.config.js:10]: Ok :v

**Pull request merged**